README.MD

```diff
@@ -1,224 +1,277 @@
+- **Write Operations:** Writes (`INSERT`, `UPDATE`, etc.) should be manually sent directly to the worker pods, bypassing the balancer.
 
-**SQL**
-```sql
-INSERT INTO manticore_cluster:idx (title) VALUES ('dog is brown'), ('mouse is small');
-```
+#### How the Balancer Works:
+1. **Dynamic Synchronization:**
+   - If `balancer.enabled` is set to `true` (default), the balancer checks the table configuration on the first active worker every 5 seconds (configurable via the `balancer.runInterval` parameter).
+   - If the configuration differs from its current state, the balancer reloads itself to align with the updated tables.
 
-**HTTP**
-```bash
-curl -X "POST" -H "Content-Type: application/x-ndjson" localhost:7308/bulk -d '
+2. **Event-Driven Updates:**
+   - When a worker starts or stops, it sends a signal to the balancer, prompting immediate synchronization without waiting for the interval.
 
-{"insert": {"cluster":"manticore_cluster", "index" : "idx", "doc": { "title" : "dog is brown" } } }
-{"insert": {"cluster":"manticore_cluster", "index" : "idx", "doc": { "title" : "mouse is small" } } }'
-```
+3. **High Availability Strategy:**
+   - The balancer achieves HA by using `index_ha_strategy: nodeads`, ensuring queries are routed only to healthy nodes.
 
-### Forward load balancer's port to your local machine:
+4. **Replica Redundancy:**
+   - To further improve HA for read operations, you can increase the number of balancer replicas by adjusting the `balancer.replicaCount` parameter in `values.yaml`.
 
-```bash
-export NAMESPACE=manticore
-export BALANCER_NAME=$(kubectl get pods --namespace $NAMESPACE \
--l "app.kubernetes.io/name=manticoresearch,app.kubernetes.io/instance=manticore,name=manticore-manticoresearch-balancer" \
--o jsonpath="{.items[0].metadata.name}")
-```
+#### Configuration Options:
+- **Custom Behavior:** You can customize the balancer’s behavior using its HA strategy (`index_ha_strategy`).
+- **Enhanced Performance:** Scaling the number of balancer replicas improves the cluster’s ability to handle read requests.
 
-| SQL | HTTP | 
-|---|---|
-|`kubectl --namespace $NAMESPACE port-forward $BALANCER_NAME 6306:9306`|`kubectl --namespace $NAMESPACE port-forward $BALANCER_NAME 6308:9308`|
+For more details, refer to the [Load Balancing documentation](https://manual.manticoresearch.com/dev/Creating_a_cluster/Remote_nodes/Load_balancing#Load-balancing).
 
-### Search through the balancer:
+### How Does Auto OPTIMIZE Work?
 
-| SQL | HTTP | 
-|---|---|
-|`mysql -h0 -P6306 -e "SELECT * FROM idx WHERE match('dog')"`|`curl -X "POST" localhost:6308/search -d '{"index" : "idx", "query": { "match": { "_all": "dog" } } }'`|
+To enable automatic index compaction, update `values.yaml` and set `optimize.enabled = true`. The `optimize.interval` parameter controls how often disk chunk merging attempts occur.
```

Comment:
Why is it disabled by default? Is it correct?

Reply:
Yes, this functionality was developed before the introduction of auto-optimization in searchd. It performs similar actions, so we have decided to disable it by default.

Reply:
The auto-optimize built in helm has some advantages.